diff --git a/dist/api.js b/dist/api.js
index 20092fb735b2494e23d358c99e122f96578a6c10..21fe588750eae24a99ae4b0396fe19c5b323c191 100644
--- a/dist/api.js
+++ b/dist/api.js
@@ -313,7 +313,7 @@ export class API {
             body.buffer = buffer;
         }
         log("body: %O", body);
-        const validator = "Input" in module && module.Input instanceof z.ZodObject ? module.Input : z.object({});
+        const validator = "Input" in module && module.Input instanceof z.ZodType ? module.Input : z.object({});
         const validation = validator.safeParse(body);
         if (!validation.success) {
             throw error(400, `Invalid body.\n${fromZodError(validation.error).message}`);
